### PR TITLE
feat: exceptions handlers

### LINKED ISSUE #2690: fix: TypeORM errors should be caught by Nest.js exception filter
## Scope & Context

TypeORM errors are not handled properly, so they're not transformed into GraphQL error.

## Current behavior

Our current implementation of exception filter is really small:

```typescript
import { Catch, UnauthorizedException } from '@nestjs/common';
import { GqlExceptionFilter } from '@nestjs/graphql';

import { GraphQLError } from 'graphql';

@Catch()
export class ExceptionFilter implements GqlExceptionFilter {
  catch(exception: Error) {
    if (exception instanceof UnauthorizedException) {
      throw new GraphQLError('Unauthorized', {
        extensions: {
          code: 'UNAUTHENTICATED',
        },
      });
    }

    return exception;
  }
}
```

## Expected behavior

We should handle more exception like so:

```typescript
import { Catch, UnauthorizedException } from '@nestjs/common';
import { GqlExceptionFilter } from '@nestjs/graphql';

import { GraphQLError } from 'graphql';
import { QueryFailedError } from 'typeorm';

@Catch()
export class ExceptionFilter implements GqlExceptionFilter {
  catch(exception: Error) {
    if (exception instanceof QueryFailedError) {
      throw new GraphQLError('Query Failed', {
        extensions: {
          code: 'QUERY_FAILED',
          exception: {
            message: exception.message,
            errorCode: exception.driverError?.code,
          },
        },
      });
    }
    if (exception instanceof UnauthorizedException) {
      throw new GraphQLError('Unauthorized', {
        extensions: {
          code: 'UNAUTHENTICATED',
        },
      });
    }

    return exception;
  }
}
```

## Technical inputs

- To optimize and avoid a big list of `if` conditions, we should rely on a mapping strategy

### LINKED ISSUE #2831: [Timebox] Investigate Sentry on Backend
Last week, we have put Sentry in place in the BE.
We already have tickets to improve exception handling on the backend. This ticket is to investigate the current behaviors and challenge the strategy regarding sending exceptions to Sentry (add user breadcrump context, make sure GraphQL exceptions that should be reported are reported)

@magrinj FYI

### PR DESCRIPTION
**Pull request by magrinj**
This PR create a new integration called `exception-capturer`, this one only contains two driver for now `Console` and `Sentry`.

The goal of this integration is to send exception to a bug tracker like Sentry.

To capture exception, new exceptions filters has been created:
- One that is capturing `HttpException` and transforming them into `GraphQLError`
- One that is capturing `TypeORMError` and transforming them into `GraphQLError`
- And the last one that is capturing all the other exceptions and sending them to the `exception-capturer`.

### CODE REVIEW

File: server/.env.example [hunk 1]
```diff
@@ -24,6 +24,7 @@ SIGN_IN_PREFILLED=true
 # SUPPORT_FRONT_HMAC_KEY=replace_me_with_front_chat_verification_secret
 # SUPPORT_FRONT_CHAT_ID=replace_me_with_front_chat_id
 # LOGGER_DRIVER=console
+# EXCEPTION_CAPTURER_DRIVER=sentry
```
Comment (charlesBochet) [inline server/.env.example:27]: EXCEPTION_HANDLER ?

File: server/src/filters/global-exception.filter.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+import { ExceptionCapturerService } from 'src/integrations/exception-capturer/exception-capturer.service';
+
+@Catch()
+@Injectable()
+export class GlobalExceptionFilter implements GqlExceptionFilter {
+  constructor(
+    private readonly exceptionCapturerService: ExceptionCapturerService,
+  ) {}
+
+  catch(exception: unknown, host: ArgumentsHost) {
```
Comment (charlesBochet) [inline server/src/filters/global-exception.filter.ts:13]: we don't use host?

**Review by charlesBochet — APPROVED** [review]
Very nice! I'm aligned with all changes

My only feedback is about the naming exception capturer. Exception Handler seems a better and more conventional naming

File: server/src/filters/global-exception.filter.ts (hunk 2 above)
Comment (magrinj) [reply server/src/filters/global-exception.filter.ts:13]: We don't really need it actually, I'll remove it

**Pull request merged**